src/bin/common/solving_loop.ml

```diff
@@ -407,8 +410,50 @@ let main () =
       | ":produce-unsat-assumptions"
       | ":print-success"
       | ":random-seed"), _
-      -> Printer.print_wrn "unsupported option %s" name
-    | _ -> Printer.print_wrn "unsupported option %s" name
+      -> unsupported_opt name
+    | _ -> unsupported_opt name
+  in
+
+  let pp_reason_unknown fmt st =
+    match State.get partial_model_key st with
+    | None ->
+      (* This case should happen only when there is no check-sat before
+         a get-info statement. *)
+      Printer.print_err "Invalid (get-info :reason-unknown)";
+      Format.fprintf fmt ""
```

Comment:
The SMT-LIB way of indicating errors is to return an sexpr of the form `(error "Blah")` (this is required to not confuse automated tools; response to all commands must go to the standard input).